File: java-football-manager-rest/src/main/java/com/radialo/footballmanager/model/Player.java
package com.radialo.footballmanager.model;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.Data;

@Data
@Entity
public class Player {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String firstName;
    private String lastName;
    private int experienceMonths;
    private int age;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "team_id")
    private Team team;
}

File: angular-football-manager/src/app/model/player.ts
export interface PlayerRequest {
  firstName: string;
  lastName: string;
  experienceMonths: number;
  age: number;
  teamId: number;
}
  
export interface PlayerResponse {
  id: number;
  firstName: string;
  lastName: string;
  experienceMonths: number;
  age: number;
  teamId: number;
}